### PR TITLE
added AI; cleanup auth actions; added tests and lint cleanup

### PR DESCRIPTION
**Pull request by nicolehaugen79**
Here's what's included with this PR:

1.) Turned on App Insights - right now, I kept things simple and am tracking all http requests and have verified that I see them being logged in Azure; note that we will need to remove the key that is stored for this before the sample is shared (we also have other keys that need to be removed throughout the app).  

2.) Cleaned up the Auth action so that it follows a more typical pattern.

3.) Added 2 integration tests - I only have these configured to run locally on a machine and have the api gateway's url hard-coded in the test for now.  My next work item is about getting these tests to run as part of CI, so I will fix the url when I figured out how to setup Jenkins and run these tests against a test environment.

4.) I added the gulp file back so that there is an easy way to run the tests that I added; also, I figured we should still have lint.  As a result of adding lint back, I also cleaned up errors\warnings that existed in the code.

### CODE REVIEW

File: apigateway/routes/cart.js [hunk 1]
```diff
@@ -28,6 +32,9 @@ router.use((req, res, next) => {
 });
 
 router.get('/api/items', function getCart(req, res) {
+
+    aiClient.trackRequest(req, res);
```
Comment (chlowell) [inline apigateway/routes/cart.js]: You can factor this into a single middleware, e.g. `router.use((req, res, next) => { aiClient.trackRequest(req, res); next(); });`

File: stickerService/itemPopularity.js [hunk 2]
```diff
@@ -20,14 +20,14 @@ const redisClient = redis.createClient({
     port: process.env.REDIS_PORT,
     password: process.env.REDIS_PASSWORD || undefined,
     tls: process.env.REDIS_TLS,
-    retry_strategy: options => {
```
Comment (chlowell) [inline stickerService/itemPopularity.js]: The `redis` client requires `retry_strategy`.

File: gulpfile.js [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+                jsx: true
+            }
+        },
+        env: {
+            node: true,
+            es6: true
+        }
+    }, eslintConfig);
+
+    // note that when we return the stream here, builds from the root (e.g., `gulp lint build`) will
```
Comment (chlowell) [inline gulpfile.js]: These lint tasks look mostly identical. Can you refactor them to share some code, and not duplicate the comments?

File: sessionService/data/sources/redisClient.js [hunk 4]
```diff
@@ -5,14 +5,14 @@ const redisClient = require('redis').createClient({
     port: process.env.REDIS_PORT,
     password: process.env.REDIS_PASSWORD || undefined,
     tls: process.env.REDIS_TLS,
-    retry_strategy: options => {
```
Comment (chlowell) [inline sessionService/data/sources/redisClient.js]: Same as the other comment: not our code, we don't choose the name.

File: tests/apigateway-tests.js [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+                res.body.items.should.have.lengthOf(1);
+
+            //Remove item from the cart
+                return app.delete('/cart/api/items/11')
+                    .query({token: userId})
+                    .set('Accept', 'application/json')
+                    .expect(200)
+                    .expect('Content-Type', /json/);
+            })
+            .then (function finishTEst(res) {
```
Comment (chlowell) [inline tests/apigateway-tests.js]: `finishTest`?

File: stickerService/itemPopularity.js (hunk 2 above)
Comment (nicolehaugen79) [inline stickerService/itemPopularity.js]: Whoops...got a little aggressive with cleaning up lint warnings...I reverted this back.

File: gulpfile.js (hunk 3 above)
Comment (nicolehaugen79) [inline gulpfile.js]: I copied these verbatim from the old gulp file...anyway, I have cleaned up the comments and also refactored the common code.

File: client/src/utils/api/auth-api.js [hunk 6]
```diff
@@ -1,14 +1,20 @@
-import { createAuthChangedAction } from '../../actions/auth-actions';
+import { createGetUserProfileRequestAction, createGetUserProfileSuccessAction, createGetUserProfileFailAction } from '../../actions/auth-actions';
```
Comment (chlowell) [inline client/src/utils/api/auth-api.js]: This is long, and you want everything that's exported; you could do something shorter like `import * as auth ...` instead.

File: gulpfile.js [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+        // of the file object so it can be used by other modules.
+        .pipe(eslint(config))
+        // eslint.format() outputs the lint results to the console.
+        // Alternatively use eslint.formatEach() (see Docs).
+        .pipe(eslint.format())
+        // To have the process exit with an error code (1) on
+        // lint error, return the stream and pipe to failAfterError last.
+        .pipe(eslint.failAfterError());
+
+    //addInputFiles(['./apigateway/**/*.js', '!./apigateway/node_modules/**/*'], config);
```
Comment (chlowell) [inline gulpfile.js]: Should this be deleted or uncommented?

Comment (nicolehaugen79) [inline gulpfile.js]: Good catch - I had commented this out when I was testing things and forgot to uncomment it.

**Pull request merged**